crates/core/src/table/builder.rs

```diff
@@ -330,6 +330,12 @@ impl DeltaTableBuilder {
         }
         Ok(table)
     }
+
+    /// Check if the [`DeltaTable`] exists
+    pub async fn verify_deltatable_existence(self) -> DeltaResult<bool> {
```

Comment:
Great work! Just minor feedback, I think this method could be on the DeltaTable struct directly

Reply:
Thanks! Indeed, having it in the `DeltaTable` struct makes more sense. Updated PR.